### ISSUE TITLE
d4rlpy MDPDataset

### ISSUE THREAD

**Issue by cclvr** [takuseno/d3rlpy#135]
Hi @takuseno, firstly thanks a lot for such a high quality repo for offline RL. 
I have a question about the method  get_d4rl(), why the rewards are all moved by one step?
        while cursor < dataset_size:
            # collect data for step=t
            observation = dataset["observations"][cursor]
            action = dataset["actions"][cursor]
            if episode_step == 0:
                reward = 0.0
            else:
                reward = dataset["rewards"][cursor - 1]

Long for your feedback.

**Comment by takuseno** [comment]
@cclvr Thanks for the issue. This is because MDPDatataset expects rewards `[r_t, r_t+1, r_t+2, ...]`, but D4RL datasets has rewards in `[r_t+1, r_t+2, r_t+3, ...]`. This is why I did something weird there.

**Comment by takuseno** [comment]
Without this change, the algorithm performance will be significantly worse.

**Comment by cclvr** [comment]
> Without this change, the algorithm performance will be significantly worse.

Thanks a lot for your reply~ 
By the way, it will be appreciated if DICE family OPE methods can be implemented here.  I know I ask too much  ^-^

**Comment by takuseno** [comment]
Thanks for the proposal! Actually, I'm working on an academic paper of this library and preparing v1.0.0 release. Once those things are done, I'll look for a way to add more OPE methods. BTW, I'm always happy for the contributions from users 😄

**Comment by mamengyiyi** [comment]
Hi, I thought the data collecting process of D4RL is the same with the commonly used method in online training, i.e., the transition is <s_t, a_t, r_t, s_t+1>. I thought the r_t indicates the reward at s_t after executing action a_t, while the reward is obtained at t+1. See code [https://github.com/rail-berkeley/d4rl/blob/master/scripts/generation/mujoco/collect_data.py](here). I'm not sure if I understand this code correctly. I'm aprreciated if you could check D4RL codes again and discuss with me further

**Comment by takuseno** [comment]
In d3rlpy, that reward is considered to be got at t+1. This function tells the implementation difference.
https://github.com/takuseno/d3rlpy/blob/747b1747ad3e41eae8a93a8e02ca02c4d9e0ccb0/d3rlpy/online/iterators.py#L471

I believe I'm following Sutton book's notation.

**Comment by takuseno** [comment]
Yes, I'm right.
https://web.stanford.edu/class/psych209/Readings/SuttonBartoIPRLBook2ndEd.pdf
![image](https://user-images.githubusercontent.com/5235131/132940374-4a0e2e8e-675a-4f4f-94a2-e7a525370b52.png)

**Comment by mamengyiyi** [comment]
Picture from Sutton's book.
<img width="565" alt="3de06b16810865514e5aafb269a6a5c" src="https://user-images.githubusercontent.com/23417331/132940680-307e197c-d7e6-483f-970d-dc307a8f6719.png">
Actually, the R_{t+1} is obtained at timestep t+1, but it indicates the reward of <s_t, a_t>.

**Comment by takuseno** [comment]
Yes, but, it's at t+1. Anyway, don't worry, the implementation is correct.
